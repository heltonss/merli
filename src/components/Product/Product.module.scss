@import '../../assets/scss/colors';
@import '../../assets/scss/mixins';

.container {
  background: #ffffff;
  display: flex;
  justify-content: space-between;
  padding: 16px;
  border-bottom: solid 1px $light-gray;
  color: $color-title;

  &:focus {
    outline: solid 0.5px $blue;
  }

  &:active,
  &:visited {
    color: $color-title;
  }

  &:last-child {
    border-bottom: none;
  }

  .image {
    display: inline-block;
    & > img {
      width: 100%;
      border-radius: 4px/4px;
    }
  }

  .info {
    margin-left: 16px;
    width: 100%;
    max-width: 380px;

    .price {
      @include fontSize(24px);
      display: block;
      color: $color-title;
      margin-bottom: 32px;
    }

    .title {
      @include fontSize(18px);
    }
  }

  .region {
    @include fontSize(12px);
    display: block;
    margin: 44px 100px 0 0;

    &:last-child {
      margin-left: auto;
    }
  }
}
