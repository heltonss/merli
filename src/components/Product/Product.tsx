import React from 'react';
import Style from './Product.module.scss';

interface IOwnProps {
  price: number;
  title: string;
  region: string;
}

const Product: React.FC<IOwnProps> = ({ price, title, region }: IOwnProps) => {
  return (
    <div className={Style.container}>
      <div className={Style.image}>
        <img src="https://via.placeholder.com/180" alt={title} />
      </div>
      <div className={Style.info}>
        <span className={Style.price}>{price}</span>
        <p className={Style.title}>{title}</p>
      </div>
      <span className={Style.region}>{region}</span>
    </div>
  );
};

export default Product;
