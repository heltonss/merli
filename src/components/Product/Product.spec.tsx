import { mount, shallow } from 'enzyme';
import React from 'react';
import Product from './Product';

describe('Product', () => {
  it('Should render without crash', () => {
    const wrapper = shallow(<Product />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should get props of the product', () => {
    const wrapper = mount(<Product price={'1980,00'} title={'Apple'} region={'sp'} />);

    expect(wrapper.props().price).toEqual('1980,00');
    expect(wrapper.props().title).toEqual('Apple');
    expect(wrapper.props().region).toEqual('sp');
  });
});
