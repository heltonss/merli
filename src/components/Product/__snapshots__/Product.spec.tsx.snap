// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Should render without crash 1`] = `ShallowWrapper {}`;
